### ISSUE TITLE
accessibility attributes role and aria-* are stripped

### ISSUE THREAD

**Issue by murraycem** [cure53/DOMPurify#203]
> sanitize should not strip out aria accessibility attributes. or a configuration should be provided that allows these attributs

### Background & Context

In order to support WCAG standards for accessibility it is necessary in some cases to added the role or any one of many aria-* attributes to the html markup. In order to guard against XSS I am using DOMPurify to sanitize the html. This is removing the aria tags so I cannot support accessibility

#### Input
`<div class="msg" role="dialog" aria-labelledby="msg--title"><button type="button" aria-label="close" class="modal-close"><i class="icon-close" ></i>some button</button></div`

#### Given output
`<div aria-labelledby="msg--title" role="dialog"><button><i></i>some button</button></div>`

#### Expected output

I expected the output to be the same as the input

`<div class="msg" role="dialog" aria-labelledby="msg--title"><button type="button" aria-label="close" class="modal-close"><i class="icon-close" ></i>some button</button></div`>

### Feature

Have the default purify configuration leave the accessibility attributes in place.
If this is not optimal provide a configuration hook to allow the attributes to pass through.

**Comment by cure53** [comment]
Heya, that request makes perfect sense and we can take care of that. 

I am wondering - do you happen to have a complete list of attributes that are needed here?

**Comment by murraycem** [comment]
See https://www.w3.org/TR/html-aria/

**Comment by cure53** [comment]
Okay, so factually `role` and `aria-*` - correct?

I would propose to allow them by default and offer a configuration flag if someone wants to remove them all. Like we do for data attributes.

```
var clean = DOMPurify.sanitize(dirty, {ALLOW_DATA_ATTR: false});
```
Makes sense?

**Comment by murraycem** [comment]
Yes role and aria-*.  Sounds great. Thanks

**Comment by cure53** [comment]
On it :)

**Comment by cure53** [comment]
I pushed a first commit to test with, feedback welcome!

**Comment by murraycem** [comment]
Looks good

**Comment by karl19** [comment]
Hey @cure53,

It is still an issue that "role" got stripped per https://github.com/cure53/DOMPurify/blob/main/src/regexp.js#L7, which i believe role should be part of it, and it works now is because "role" passed https://github.com/cure53/DOMPurify/blob/main/src/regexp.js#L8 unintentionally i guess, but whenever app who uses its own ALLOWED_URI_REGEXP, then gone the "role".

Can we have a look?

Thanks,
Karl

**Comment by cure53** [comment]
> Hey @cure53,
> 
> It is still an issue that "role" got stripped per https://github.com/cure53/DOMPurify/blob/main/src/regexp.js#L7, which i believe role should be part of it, and it works now is because "role" passed https://github.com/cure53/DOMPurify/blob/main/src/regexp.js#L8 unintentionally i guess, but whenever app who uses its own ALLOWED_URI_REGEXP, then gone the "role".
> 
> Can we have a look?
> 
> Thanks,
> Karl

Sorry, what is the problem you are referring to? I am having trouble parsing this.

**Comment by karl19** [comment]
thanks for the prompt response @cure53, basically the issue am having is the "role" attribute is stripped when you pass your own ALLOWED_URI_REGEXP config like below.
```
DOMPurify.sanitize(content, {
    ALLOWED_URI_REGEXP: /^(https?|[^a-z]|[a-z+\-.]+(?:[^a-z+.\-:]))/i,
})
```
My understanding is it should have been part of ARIA_ATTR regex, right now it is not part of it but still passing through is it passes default IS_ALLOWED_URI regex.

**Comment by cure53** [comment]
Ah, well - that is a rather harsh regex :D May I ask, what is the use-case here? _Usually_ the regex config is not needed as it is really omnipotent - maybe there is a more elegant way to make things work as you intend?
